src/user_type_field_iterator.cpp

```diff
@@ -25,5 +25,6 @@ bool UserTypeFieldIterator::next() {
     return false;
   }
   current_ = next_++;
-  return decoder_.decode_value(current_->type, value_);
+  value_ = decoder_.decode_value(current_->type);
+  return !!value_.data_type();
```

Comment:
What do you think about using the following for these checks?

```cpp
return value.is_valid();
```

where `is_valid()` is defined as:
```cpp
bool Value::is_valid() {
   return value_.data_type();
}
```

Reply:
looks good, it is more intuitive, will do the change

Reply:
done